spec/requests/api/v1/accounts/statuses_spec.rb

```diff
@@ -109,15 +105,19 @@
       end
 
       it 'returns http success' do
-        get :index, params: { account_id: account.id, pinned: true }
+        get "/api/v1/accounts/#{account.id}/statuses", params: { pinned: true }, headers: headers
+
         expect(response).to have_http_status(200)
       end
 
       context 'when user does not follow account' do
         it 'lists the public status only' do
-          get :index, params: { account_id: account.id, pinned: true }
-          json = body_as_json
-          expect(json.map { |item| item[:id].to_i }).to eq [status.id]
+          get "/api/v1/accounts/#{account.id}/statuses", params: { pinned: true }, headers: headers
+
+          expect(body_as_json)
+            .to contain_exactly(
+              include(id: status.id.to_s)
```

Comment:
I think it would be more readable as:
```suggestion
              a_hash_including(id: status.id.to_s)
```